packages/jest-util/src/CustomConsole.js

```diff
@@ -41,11 +55,29 @@ export default class CustomConsole extends Console {
     super.log(message);
   }
 
+  _storeInBuffer(type: LogType, message: string) {
+    let origin = '';
+
+    if (this._getSourceMaps) {
+      const callsite = getCallsite(3, this._getSourceMaps());
+      origin = callsite.getFileName() + ':' + callsite.getLineNumber();
+    }
+
+    this._buffer.push({message, origin, type});
```

Comment:
that might work - bigger code change, though. Also, any methods we expose on the logger (such as `on` from `EventEmitter`) will be visible on the user's global `console`.

A bigger refactor of the `console` abstraction to something event driven might make sense though

Reply:
Fair point. I just feel like this is _very_ hacky and we need to comment the 💩 out of it and refactor later :D Maybe using Proxy would be a tad cleaner?

Reply:
Proxy is a great idea!

I still think a more general refactor of the logging system (to solve this, #6441 and maybe #6524) is best left to a separate PR, but I can give it a quick go

Reply:
`Proxy` doesn't do in-place replacement, so I'd have to either replace the function at instantiation time, or in the same place.

However, I changed to mocking out `_log` on the `console`s instead of messing with the buffer. Further away from #6441, but way cleaner diff here